include/fmt/format.h

```diff
@@ -1582,79 +1586,74 @@ auto write_escaped_cp(OutputIt out, const find_escape_result<Char>& escape)
   auto c = static_cast<Char>(escape.cp);
   switch (escape.cp) {
   case '\n':
-    *out++ = '\\';
-    c = 'n';
+    *out++ = static_cast<Char>('\\');
+    c = static_cast<Char>('n');
     break;
   case '\r':
-    *out++ = '\\';
-    c = 'r';
+    *out++ = static_cast<Char>('\\');
+    c = static_cast<Char>('r');
     break;
   case '\t':
-    *out++ = '\\';
-    c = 't';
+    *out++ = static_cast<Char>('\\');
+    c = static_cast<Char>('t');
     break;
   case '"':
     FMT_FALLTHROUGH;
   case '\'':
     FMT_FALLTHROUGH;
   case '\\':
-    *out++ = '\\';
+    *out++ = static_cast<Char>('\\');
     break;
   default:
     if (is_utf8()) {
       if (escape.cp < 0x100) {
-        return format_to(out, FMT_STRING("\\x{:02x}"), escape.cp);
+        return write_codepoint<2>(out, static_cast<Char>('x'), escape.cp);
```

Comment:
Let's pass prefix as char to have the cast in one place (`write_codepoint`) instead of each call site.